src/excmds.ts

```diff
@@ -4092,10 +4092,15 @@ export async function tab_helper(interactive: boolean, anyWindow: boolean, ...ke
  * added in a future release will be SILENTLY overridden. Aliases are
  * expanded recursively.
  *
+ * If the `--url` option is given, the command will only be defined in the
+ * specified url pattern (just like `bindurl`). It is useful to overwrite
+ * command like urlparent or urlroot in some websites like github repositories.
```

Comment:
I had never thought of doing this, this is a brilliant idea :)

Reply:
In fact, there are some problems for `urlparent`. If we `alias urlparent urlparent 2`, it will be expand recursively and infinitely. I'm not sure whether should we really add this feature.

Reply:
We could disallow loops in aliases and instead persuade people to do e.g. `:alias urlparent js tri.excmds.urlparent(2)`